library/src/tests.rs

```diff
@@ -0,0 +1,181 @@
+/// account so that results of calculations can also be compared.
+pub fn test_expression(expr: &str, expected: &Value) {
+    test_expression_with_lib(expr, "", expected);
+}
+
+pub fn test_expression_with_lib(expr: &str, lib: &str, expected: &Value) {
+    test_expression_with_lib_and_profile(expr, lib, Profile::Unrestricted, expected);
+}
+
+pub fn test_expression_with_lib_and_profile(
+    expr: &str,
+    lib: &str,
+    profile: Profile,
+    expected: &Value,
+) {
+    let mut stdout = vec![];
+    let mut out = GenericReceiver::new(&mut stdout);
+
+    let sources = SourceMap::new([("test".into(), lib.into())], Some(expr.into()));
+
+    let mut interpreter =
+        stateful::Interpreter::new(true, sources, PackageType::Exe, profile.into())
+            .expect("test should compile");
+    let result = interpreter
+        .eval_entry(&mut out)
+        .expect("test should run successfully");
+
+    match (&expected, result) {
+        (&Value::Tuple(tup1), Value::Tuple(tup2)) if tup1.len() == tup2.len() => {
+            // If both values are tuples of the same length, we crack them open and compare elements
+            for (value1, value2) in tup1.iter().zip(tup2.iter()) {
+                if let (Value::Double(double1), Value::Double(double2)) = (value1, value2) {
+                    // If both elements are doubles, we use approximate comparison
+                    assert_doubles_almost_equal(*double1, *double2);
+                } else {
+                    assert_eq!(value1, value2);
+                }
+            }
+        }
+        (&Value::Double(double1), Value::Double(double2)) => {
+            assert_doubles_almost_equal(*double1, double2);
+        }
+        (&expected, result) => assert_eq!(expected, &result),
+    }
+}
+
+/// # Panics
+///
+/// Will panic if f64 values are significantly different.
+fn assert_doubles_almost_equal(val1: f64, val2: f64) {
+    let val1_abs = val1.abs();
+    let val2_abs = val2.abs();
+    if val1_abs < f64::MIN_POSITIVE && val2_abs < f64::MIN_POSITIVE {
+        // Note, that f64::MIN_POSITIVE is not the smallest representable positive number.
+        return;
+    }
+    assert!(
+        ((val1 - val2).abs() / (val1_abs + val2_abs)) < 1e-15,
+        "Significant difference between expected and actual values: val1={val1}, val2={val2}."
+    );
+}
+
+//
+// Core namespace
+//
+#[test]
+fn check_repeated() {
+    test_expression("Repeated(Zero, 0)", &Value::Array(vec![].into()));
+    test_expression(
+        "Repeated(One, 1)",
+        &Value::Array(vec![Value::RESULT_ONE].into()),
+    );
+    test_expression(
+        "Repeated(1, 2)",
+        &Value::Array(vec![Value::Int(1), Value::Int(1)].into()),
+    );
+    test_expression(
+        "Repeated(true, 3)",
+        &Value::Array(vec![Value::Bool(true), Value::Bool(true), Value::Bool(true)].into()),
+    );
+}
+
+#[test]
+fn check_exp_with_cnot() {
```

Comment:
Isn't exp part of the `Microsoft.Quantum.Intrinsic` namespace? Maybe this tests belong in the same file that other intrinsic callables are tested.

Reply:
I didn't change the effective location of those tests. If they make sense in another file, we can move them in a follow up PR.